hedera-node/src/main/java/com/hedera/services/store/tokens/HederaTokenStore.java

```diff
@@ -734,6 +791,14 @@ public ResponseCodeEnum update(TokenUpdateTransactionBody changes, long now) {
 				token.setName(newName);
 			}
 			if (changes.hasTreasury() && !changes.getTreasury().equals(token.treasury().toGrpcAccountId())) {
+				if (token.tokenType().equals(TokenType.NON_FUNGIBLE_UNIQUE)) {
+					var relationship = asTokenRel(changes.getTreasury(), tId);
+					long balance = (long) tokenRelsLedger.get(relationship, TOKEN_BALANCE);
+					if (balance != 0) {
+						appliedValidity.set(NOT_SUPPORTED);
+						return;
+					}
+				}
```

Comment:
I think that this check must be performed before line `750`. All validations are there and from line 750 down we are applying the new updates. If we revert here we won't revert the token changes